### ISSUE TITLE
Render bicycle road/cyclestreet blue independent of motor vehicle access

### ISSUE THREAD

**Issue by ottomei** [cyclosm/cyclosm-cartocss-style#666]
As dedicated bicycle infrastructure, bicycle roads and cyclestreets are generally rendered in blue similar to cycleways, clearly highlighting them compared to regular streets. However, if a bicycle road has restricted or no motor vehicle access, the bicycle road/cyclestreets is visually "downgraded" to a regular street with low or no car traffic, i.e. in a green shade. 

This creates the paradox situation where dedicated cycling infrastructure without motor vehicles is seemingly portrayed as "less preferable" for cyclists than a bicyle road/cyclestreet that allows full motor vehicle access. 

True, this is not a big issue for most bicycle roads/cyclestreets as they generally do provide full motor vehicle access. At least in Germany, however, legally speaking, motor vehicle access for bicycle roads is the exception, not the rule. And municipalities are starting to implement the rule. For example, Munich built its first "true" bicycle road without motor vehicle access: https://www.openstreetmap.org/way/504565991 But the green render doesn't show the high quality of the dedicated cycling infrastructure.

**Comment by Phyks** [comment]
In CyclOSM, blue is for bike and green is for pedestrians. Depending on what you are looking for, green might be preferable (since for sure there will not be any cars).

I can't find any pictures on the ground for what this "true" bicycle road looks like. Would you have some?

**Comment by ottomei** [comment]
The street has only recently been constructed and works haven't terminated fully (but it can already be used). On the project plan the street is shown with a sidewalk (beige) and a separate "traffic" lane (brown) made out of asphalt. Cars are excluded from the traffic lane both legally (designation as a bicycle road according to German law) and physically via access barriers at intersections. In CyclOSM, Green indicates a space reserved for pedestrians, but in this case they are supposed to use the sidewalk. Blue indicates dedicated cycling infrastructure, which in this case this legally and physically is. 
![image](https://github.com/cyclosm/cyclosm-cartocss-style/assets/157621281/64ab0996-b656-44b1-af9f-4152022961ef)
This separation can also be seen on the left side of this render:
![image](https://github.com/cyclosm/cyclosm-cartocss-style/assets/157621281/026e1f06-73da-46e2-999f-59bdff0cebc9)

But apart from this specific case, it still does not seem logical to me why a bicycle road/cyclestreet would suddenly become a "green" pedestrian mixed space only because cars have been excluded from the traffic lane. And visually also, it makes a "blue" bicycle road mixed with cars look preferable from a cyclists perspective over a "green" non-mixed bicycle road without cars.

**Comment by Phyks** [comment]
Sorry, I mixed cycle streets and pedestrian areas in my previous answer. I get your point.

Would you have an alternative render in mind?

My concern so far is that a cycle street has very different implementations. The one you are referring to cannot have any cars. But the regular belgian cars is a small increment on a 30km/h street for example: https://wiki.openstreetmap.org/wiki/Key:cyclestreet#Belgium. The latter one being rendered in blue: https://www.cyclosm.org/#map=19/50.83686/4.34681/cyclosm.

**Comment by gwater** [comment]
I agree that German "Fahrradstraßen" should absolutely be rendered in blue, similar to cycleways. They are dedicated to bicycles and most commonly cars are restricted to something like `vehicle=destination`. In most cases Fahradstraßen are much prefereable infrastructure to, e.g., a narrow cycle track on a major road. It should definitely stand out more in the CyclOSM map.

**Comment by gwater** [comment]
Notably, this road with `bicycle_road=yes` + `vehicle=no` + `motorcar=yes` is already rendered in blue in the *status quo*.

https://www.openstreetmap.org/way/22955668

**Comment by gwater** [comment]
Two more examples: These two streets have almost the same rules. On

**Comment by gwater** [comment]
Two more examples

Road 1) https://www.openstreetmap.org/way/249499075

Road 2) https://www.openstreetmap.org/way/713576589

Road 1 is tagged with `motor_vehicle=yes` which means cars (and even trucks/lorries!) can freely use this road.

Road 2 is tagged with `vehicle=destination` which excludes many cars and trucks from using the road.

Strangely, Road 2 (the one which excludes cars!) is rendered in light green and Road 1 (which is much worse for bicycles) is rendered in deep blue.

**Comment by gwater** [comment]
Is it possible that this happens because German mappers use `bicycle_road` instead of `cyclestreet`. I cannot find any mention of `bicycle_road` in https://github.com/cyclosm/cyclosm-cartocss-style/blob/master/roads.mss

**Comment by gwater** [comment]
I have put in PR #668 to address this issue

**Comment by gwater** [comment]
I played around a bit in Kosmtik and apparently my PR did not resolve this completely.

There are two issues:
1. In master branch all cyclestreets have the color `@mixed-cycle-fill`. But where we have `[motor_vehicle='no']` the color should be `@cycle-fill`.
2. In master branch cyclestreets (which are typically residential streets) become very thin on zoom levels < 14. But when these are `[motor_vehicle='no']` they should render at twice the width of a one-way cycle track because that is the functionality they provide.

With these two adjustments, car-free cyclestreets would really have appropriate visibility on the map.
